pkg/types/kustomization.go

```diff
@@ -376,3 +381,39 @@ type Replica struct {
 	// The number of replicas required.
 	Count int64 `json:"count,omitempty" yaml:"count,omitempty"`
 }
+
+// Patch represent either a Strategic Merge Patch or a JSON patch
+// and its targets.
+// The content of the patch can either be from a file
+// or from an inline string.
+type Patch struct {
+	// Path is a relative file path to the patch file.
+	Path string `json:"path,omitempty" yaml:"path,omitempty"`
+
+	// Patch is the content of a patch.
+	Patch string `json:"patch,omitempty" yaml:"patch,omitempty"`
+
+	// Target points to the resources that the patch is applied to
+	Target Target `json:"target,omitempty" yaml:"target,omitempty"`
+
+	// Type is one of `StrategicMergePatch` or `JsonPatch`.
+	// optional
```

Comment:
Type is one of "StrategicMerge" or "Json6902".
(Patch suffix redundant here)

I think this field should be required, so the kustomization is self-documenting.

Reply:
When we discussed this in the sig-cli meeting, the type of the patch can be determined from the content of the patch file.

Reply:
Good idea, just detect it.  Drop the Type field.